### ISSUE TITLE
[Serializer] Alias for attributes

### ISSUE THREAD

**Issue by estahn** [symfony/symfony#15171]
It would be very helpful to have the possibility to specify an alias for an attribute. This is the same as `serialized_name` in JMSSerializer". 

```
@SerializedName
This annotation can be defined on a property to define the serialized name for a property. If this is not
defined, the property will be translated from camel-case to a lower-cased underscored name, e.g.
camelCase -> camel_case.
```

Here is an example YAML:

``` yml
AppBundle\Entity\Product:
    attributes:
        id:
            groups: [default]
        name:
            groups: [default]
        brand.name:
            alias: brand
            groups: [default]
```

I wonder about the general take on this.

This was also mentioned in #14924 .
## 

My current workaround is using a custom NameConverter.

``` php
use Symfony\Component\Serializer\Exception\UnsupportedException;
use Symfony\Component\Serializer\NameConverter\NameConverterInterface;

class ArrayNameConverter implements NameConverterInterface
{
    protected $map = [];

    /**
     * @param array $map
     */
    public function __construct(array $map)
    {
        $this->map = $map;
    }

    public function normalize($propertyName)
    {
        return isset($this->map[$propertyName]) ? $this->map[$propertyName] : $propertyName;
    }

    public function denormalize($propertyName)
    {
        throw new UnsupportedException('This method needs to be implemented.');
    }
}

$normalizer = new ObjectNormalizer($classMetadataFactory, new ArrayNameConverter(['brand.name' => 'brand']));
```

**Comment by dunglas** [comment]
I've started a POC about that. It's very similar to what you describe. It uses a new name converter that `alias` (was named `serializedName` in my POC but `alias` looks better) metadata.

**Comment by estahn** [comment]
@dunglas Kevin, do you want me to contribute the code? It should be fairly simple to implement i assume.

**Comment by IndraGunawan** [comment]
any news for this feature?

**Comment by magarzon** [comment]
@IndraGunawan the feature was in development by @dunglas in #15200, but was close in favor of #18016, that for me is not the same

**Comment by stof** [comment]
@magarzon #15200 was not mergeable, due to breaking BC.

**Comment by SebastianLeuters** [comment]
Hey guys, 
whats the last state of this feature request? 
It would be nice to implement an "alias" annotation to customise the property names for different serialization groups.

Best regards.

**Comment by fbourigault** [comment]
@javiereguiluz this one could be closed since #28505 is merged and released.

**Comment by javiereguiluz** [comment]
Closing as fixed then. Thanks!
